### PR TITLE
Default feature flags as an object

### PR DESCRIPTION
**Pull request by mstrasinskis**
When an expected flag is not provided by `environment.featureFlags`, the missing value is taken from a [string](https://github.com/dfinity/nns-dapp/blob/662509773a569ec966ce7cfcf2ee437ff1f55e9a/frontend/src/lib/constants/environment.constants.ts#L32) that is supposed to contain all default values. The issue is that this approach cannot be verified, and it already omits some values, which could lead to a broken app state when environment constants are missing or outdated.

In cases where the `featureFlags` environment variable is empty or not provided, the dapp breaks with a dev console error: `Uncaught (in promise) Error: Missing mandatory environment variables: featureFlags`
This error is self-explanatory, so no changes are needed to handle this specific case.

However, when there are missing entries in the environment variable (e.g., `record { 0="FEATURE_FLAGS"; 1="{}" };`), the dapp also breaks with the error:`Error: derived() expects stores as input, got a falsy value.`
This happens because no derived store is created for the missing value(s). While we could modify the store creation mechanism to always initialize with the `false` value, I believe having explicitly set default values is a better solution.

# Changes

- Add `defaultFeatureFlagValues: FeatureFlags<boolean>`.
- Replace missing flags with the entries from `defaultFeatureFlagValues`.

# Tests

- Added a unit test to verify that the default flags are used.
- Tested manually: reproduced the issue with the current nns-dapp and missing flags, then confirmed that the updated version works with the default flags.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/lib/constants/environment.constants.ts [hunk 1]
```diff
@@ -19,18 +19,36 @@ export interface FeatureFlags<T> {
...
+  TEST_FLAG_NOT_EDITABLE: false,
+};
 
 export type FeatureKey = keyof FeatureFlags<boolean>;
 
+const getFeatureFlagsFromEnv = (): FeatureFlags<boolean> => {
+  let featureFlags = {};
+  try {
+    featureFlags = JSON.parse(envVars?.featureFlags);
+  } finally {
```
Comment (yhabib) [inline frontend/src/lib/constants/environment.constants.ts:38]: why not catch the error? if we don't expect any errors then we don't need the `try`

File: frontend/src/lib/constants/environment.constants.ts [hunk 2]
```diff
@@ -19,18 +19,36 @@ export interface FeatureFlags<T> {
...
+  ENABLE_EXPORT_NEURONS_REPORT: false,
+  TEST_FLAG_EDITABLE: false,
+  TEST_FLAG_NOT_EDITABLE: false,
+};
 
 export type FeatureKey = keyof FeatureFlags<boolean>;
 
+const getFeatureFlagsFromEnv = (): FeatureFlags<boolean> => {
+  let featureFlags = {};
+  try {
```
Comment (dskloetd) [inline frontend/src/lib/constants/environment.constants.ts:36]: What's the purpose of this `try`?
With an empty `finally` block and no `catch` block, to the best of my knowledge, this does nothing and can be removed.

File: frontend/src/tests/lib/constants/environment.constants.spec.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { defaultFeatureFlagValues } from "$lib/constants/environment.constants";
+import * as envVarsUtils from "$lib/utils/env-vars.utils";
+
+describe("FEATURE_FLAG_ENVIRONMENT", () => {
+  const environmentVars = envVarsUtils.getEnvVars();
+
+  beforeEach(() => {
+    vi.resetModules();
```
Comment (dskloetd) [inline frontend/src/tests/lib/constants/environment.constants.spec.ts:8]: Can you add a comment explaining what `vi.resetModules()` is for?

File: frontend/src/lib/constants/environment.constants.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/constants/environment.constants.ts]: Totally agree, updated.

File: frontend/src/tests/lib/constants/environment.constants.spec.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/constants/environment.constants.spec.ts:11]: Done.

File: frontend/src/lib/constants/environment.constants.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/constants/environment.constants.ts:36]: Good point, the catch has been added.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/constants/environment.constants.ts [hunk 4]
```diff
@@ -19,18 +19,36 @@ export interface FeatureFlags<T> {
   TEST_FLAG_EDITABLE: T;
   TEST_FLAG_NOT_EDITABLE: T;
 }
+export const defaultFeatureFlagValues: FeatureFlags<boolean> = {
+  ENABLE_CKBTC: true,
```
Comment (yhabib) [inline frontend/src/lib/constants/environment.constants.ts:23]: You need to delete this one as https://github.com/dfinity/nns-dapp/pull/5822 in the queue. sorry 😬

Comment (mstrasinskis) [reply frontend/src/lib/constants/environment.constants.ts:23]: Sure thing! Thanks for the heads-up.

**Review by yhabib — APPROVED** [review]

File: frontend/src/tests/lib/constants/environment.constants.spec.ts [hunk 5]
```diff
@@ -27,4 +27,20 @@ describe("FEATURE_FLAG_ENVIRONMENT", () => {
...
+    vi.spyOn(envVarsUtils, "getEnvVars").mockReturnValue({
+      ...environmentVars,
+      featureFlags: `{"TEST_FLAG_NOT_EDITABLE": TRUE}`,
+    });
+
+    const { FEATURE_FLAG_ENVIRONMENT } = await import(
+      "$lib/constants/environment.constants"
+    );
+    expect(FEATURE_FLAG_ENVIRONMENT).toEqual(defaultFeatureFlagValues);
+    expect(spyConsoleError).toBeCalledTimes(1);
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/tests/lib/constants/environment.constants.spec.ts:47]: Good point. Updated.

**Pull request merged**